HotelBase/HotelFactory.cs

```diff
@@ -17,5 +18,10 @@ public class HotelFactory : IHotelFactory
                     return null;
             }
         }
+
+        public List<int> PresentAvailableIds()
+        {
+            return new List<int> {1,2};
```

Comment:
Should have:
 
In two places in your Factory class you have define HotelIds. it will be good to have this list or whatever defined once and used in both places